### PR TITLE
Isolate merge tree to not export in the future

### PR DESCRIPTION
**Pull request by anthony-murphy**
## Description
This change has no functional changes. It isolates the merge tree class and deprecates it, which will allow us to remove it from the exports in the next major release. The primary interface for merge tree is actually the client class, which is unchanged here. Overall, this will increase our ability to make changes to underlying data structure, MergeTree with needing breaking changes.

### CODE REVIEW

File: packages/dds/merge-tree/src/MergeTreeTextHelper.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    return accum.parallelArrays === true;
+}
+
+type ITextAccumulatorType = ITextAccumulator | ITextAndMarkerAccumulator;
+
+/**
+ * @deprecated  for internal use only. public export will be removed.
+ * @internal
+ */
+export class MergeTreeTextHelper implements IMergeTreeTextHelper {
```
Comment (anthony-murphy) [inline packages/dds/merge-tree/src/MergeTreeTextHelper.ts:36]: all copy pasted from text segment. added an interface so we don't need to export this class which is coupled to merge tree

File: packages/dds/merge-tree/src/mergeTree.ts [hunk 2]
```diff
@@ -28,92 +28,66 @@ import {
      LocalReferenceCollection,
      LocalReferencePosition,
 } from "./localReference";
+import {
+	BaseSegment,
```
Comment (anthony-murphy) [inline packages/dds/merge-tree/src/mergeTree.ts:32]: no real changes here. just moves between this file and mergeTreeNodes to minimize export surface

File: BREAKING.md [hunk 3]
```diff
@@ -35,6 +36,9 @@ to be `string | undefined`.
 ### `IContainerRuntime.flush` is deprecated
 `IContainerRuntime.flush` is deprecated and will be removed in a future release. If a more manual flushing process is needed, move all usage to `IContainerRuntimeBase.orderSequentially` if possible.
 
+### MergeTree class is deprecated
+    The MergeTree class is deprecated and will no longer be exported in the next release. This should no affect usage as MergeTree is an internal class, and the public API exists on the Client class, which will continue to be exported and supported.
```
Comment (scarlettjlee) [inline BREAKING.md:40]: Typo: "no**t** affect usage"

**Review by scarlettjlee — APPROVED** [review]

File: BREAKING.md (hunk 3 above)
Comment (anthony-murphy) [reply BREAKING.md:40]:
```suggestion
    The MergeTree class is deprecated and will no longer be exported in the next release. This should not affect usage as MergeTree is an internal class, and the public API exists on the Client class, which will continue to be exported and supported.
```

**Pull request merged**